Code/GraphMol/FileParsers/ProximityBonds.cpp

```diff
@@ -24,12 +24,14 @@ static const double MAXDIST2 = 29.7025;  // MAXDIST*MAXDIST
 
 struct ProximityEntry {
   float x, y, z, r;
-  int atm, hash, next;
+  int atm, hash, next, elem;
 
   bool operator<(const ProximityEntry &p) const { return x < p.x; }
 };
 
-static bool IsBonded(ProximityEntry *p, ProximityEntry *q) {
+static bool IsBonded(ProximityEntry *p, ProximityEntry *q, unsigned int flags) {
+  if (flags & ctdIGNORE_H_H_CONTACTS && p->elem == 1 && q->elem == 1)

```

Comment:
After looking at the code and thinking about this a bit I changed my mind.
This is a code path for determining whether or not atoms should be bonded, not doing sanity checks.
We could think about adding a structure "linter" that detects suspicious distances and angles, but that would be a different feature.